docs/Release_Management.md

```diff
@@ -0,0 +1,49 @@
+# Release Management
+
+## Overview
+This document describes **Azure Key Vault Provider for Secrets Store CSI Driver** project release management, which talks about versioning, branching and cadence.
+
+## Legend
+
+- **X.Y.Z** refers to the version (git tag) of AKV provider for Secrets Store CSI Driver that is released. This is the version of the AKV provider for Secrets Store CSI Driver image.
+ 
+- **Milestone** should be designed to include feature sets to accommodate monthly release cycles including test gates. GitHub milestones are used by maintainers to manage each release. PRs and Issues for each release should be created as part of a corresponding milestone.
+
+- **Test gates** should include soak tests and upgrade tests from the last minor version.
+
+
+## Versioning
+This project strictly follows [semantic versioning](https://semver.org/spec/v2.0.0.html). All releases will be of the form _vX.Y.Z_ where X is the major version, Y is the minor version and Z is the patch version.
```

Comment:
The current set of releases don't have the version prefix `v`. We should use that in the next release: `v0.1.0`